utils/common/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -110,6 +110,22 @@ fun File.unpack(
         }
     }
 
+/**
+ * Try to unpack this [File] of an unknown archive type to [targetDirectory] using [filter] to select only the entries
+ * of interest. Use the file name to guess the archive type. If this is not possible due to a missing or unknown file
+ * extension, try out the supported archive types one after the other.
+ */
+fun File.unpackTryAllTypes(targetDirectory: File, filter: (ArchiveEntry) -> Boolean = { true }) {
+    (listOf(ArchiveType.getType(this.name)) + ArchiveType.values().toList()).distinct().asSequence()
+        .filterNot { it == ArchiveType.NONE }
+        .firstOrNull { archiveType ->
```

Comment:
Use `find` instead of `firstOrNull`.

Reply:
Overall proposal:

    val typeFromName = ArchiveType.getType(name)

    enumValues<ArchiveType>().mapNotNullTo(mutableListOf(typeFromName)) { type ->
        type.takeUnless { type == typeFromName || type == ArchiveType.NONE }
    }.find { type ->
        // ...
    }

Reply:
Changed accordingly.